opentracing/span.py

```diff
@@ -208,7 +203,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 'python.exception.val': exc_val,
                 'python.exception.tb': exc_tb,
                 })
-        self.finish()
+        self.deactivate()
```

Comment:
the context manager never finishes the `Span`; it's the `Span.deactivate()` that may call the finish so:
* default `with` case -> `finish()` is always called
* if a `capture()` has been invoked, the `Span` must be activated and deactivated another time otherwise it will not be finished